src/seat.c

```diff
@@ -189,11 +189,21 @@ configure_libinput(struct wlr_input_device *wlr_input_device)
 		wlr_log(WLR_INFO, "dwt configured");
 		libinput_device_config_dwt_set_enabled(libinput_dev, dc->dwt);
 	}
-	if (libinput_device_config_click_set_method(libinput_dev,
-			dc->click_method) > 0) {
+	if (dc->click_method < 0
+			||
+			(libinput_device_config_click_get_methods(libinput_dev)
+				& dc->click_method) == 0) {
```

Comment:
- [ ] should I turn this around? put the `< 0` at the end?

It makes the logic of available click methods a bit weird. I think -1 would be `0b11111...1111`. So the check with the flag would probably always succeed in that case.

Reply:
I think the whole if could be turned around, something like this (not tested at all)
```c
if (dc->click_method > 0
		&& libinput_device_config_click_get_methods(libinput_dev) & dc->click_method) {
	libinput_device_config_click_set_method(..);
} else {
	wlr_log(WLR_INFO, "click method not configured");
}
```

Reply:
I just turned it around now. In my head this is most similar to the existing code: In the existing code, the checking whether an option is supported is already done in the same way;  -1 is also passed in that case.